### ISSUE TITLE
The app crash on the last beta 

### ISSUE THREAD

**Issue by yotam20** [kyujin-cho/pixel-volte-patch#225]
- Device: pixel 6
 - OS: AP11.231215.007

**Comment by Demonic16** [comment]
Same to me. Crashes only when Shizuku is running. If Shizuku is stopped then Pixel IMS launches but obviously can't do anything

**Comment by iKirby** [comment]
This is because in the latest beta the following method changed. They added a third parameter which is a boolean.

```
List<SubscriptionInfo> getActiveSubscriptionInfoList(String str, String str2, boolean z) throws RemoteException;
```

The source code of beta versions won't be released until it reaches stable. There is no documentation and I don't know what the boolean means. 

I haven't used hidden apis in Android Studio. So I modified Pixel IMS using APKTool.

The only file I modified is CarrierModer.smali, starting at line 310.

```
const/4 v1, 0x0

.line 81
invoke-interface {v0, p0, p0, v1}, Lcom/android/internal/telephony/ISub;->getActiveSubscriptionInfoList(Ljava/lang/String;Ljava/lang/String;Z)Ljava/util/List;
```

Passing false for the third parameter works (at least for now).

If you need it you can download it below. GitHub doesn't allow uploading apk files so I put it in a zip archive.

[pixelims_zipaligned.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/13896231/pixelims_zipaligned.zip)

or get it on Google Drive: https://drive.google.com/file/d/1zqVShKvd9adwXSFVB1yVh81jjcX4dagB/view?usp=sharing

**Comment by Aduhelm** [comment]
> This is because in the latest beta the following method changed. They added a third parameter which is a boolean.
> 
> ```
> List<SubscriptionInfo> getActiveSubscriptionInfoList(String str, String str2, boolean z) throws RemoteException;
> ```
> 
> The source code of beta versions won't be released until it reaches stable. There is no documentation and I don't know what the boolean means.
> 
> I haven't used hidden apis in Android Studio. So I modified Pixel IMS using APKTool.
> 
> The only file I modified is CarrierModer.smali, starting at line 310.
> 
> ```
> const/4 v1, 0x0
> 
> .line 81
> invoke-interface {v0, p0, p0, v1}, Lcom/android/internal/telephony/ISub;->getActiveSubscriptionInfoList(Ljava/lang/String;Ljava/lang/String;Z)Ljava/util/List;
> ```
> 
> Passing false for the third parameter works (at least for now).
> 
> If you need it you can download it below. GitHub doesn't allow uploading apk files so I put it in a zip archive.
> 
> [pixelims_zipaligned.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/13896231/pixelims_zipaligned.zip)
> 
> or get it on Google Drive: https://drive.google.com/file/d/1zqVShKvd9adwXSFVB1yVh81jjcX4dagB/view?usp=sharing

IT works

**Comment by yotam20** [comment]
It works for me thanks

**Comment by ofir29200** [comment]
> This is because in the latest beta the following method changed. They added a third parameter which is a boolean.
> 
> ```
> List<SubscriptionInfo> getActiveSubscriptionInfoList(String str, String str2, boolean z) throws RemoteException;
> ```
> 
> The source code of beta versions won't be released until it reaches stable. There is no documentation and I don't know what the boolean means. EDIT: the third parameter is named `isForAllProfiles`, probably user profiles?
> 
> I haven't used hidden apis in Android Studio. So I modified Pixel IMS using APKTool.
> 
> The only file I modified is CarrierModer.smali, starting at line 310.
> 
> ```
> const/4 v1, 0x0
> 
> .line 81
> invoke-interface {v0, p0, p0, v1}, Lcom/android/internal/telephony/ISub;->getActiveSubscriptionInfoList(Ljava/lang/String;Ljava/lang/String;Z)Ljava/util/List;
> ```
> 
> Passing false for the third parameter works (at least for now).
> 
> If you need it you can download it below. GitHub doesn't allow uploading apk files so I put it in a zip archive.
> 
> [pixelims_zipaligned.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/13896231/pixelims_zipaligned.zip)
> 
> or get it on Google Drive: https://drive.google.com/file/d/1zqVShKvd9adwXSFVB1yVh81jjcX4dagB/view?usp=sharing
> 

It doesn't work for me.
It can see that shizuku is running, but it thinks it doesn't have permission, but in shizuku I can see that it does have permission

![Screenshot_20240111-072135.png](https://github.com/kyujin-cho/pixel-volte-patch/assets/14427360/60b9e48b-6e7f-47b8-837b-1f74b9fd489d)

![Screenshot_20240111-072129.png](https://github.com/kyujin-cho/pixel-volte-patch/assets/14427360/2a2b8a30-7e1b-453e-9315-51a33993b355)

**Comment by iKirby** [comment]
@ofir29200 Could you please check if you have the latest version of Shizuku installed (v13.5.3 as of now)? I did just modify the two lines of code I've posted, that shouldn't affect other parts of the app.

**Comment by hsoc** [comment]
> This is because in the latest beta the following method changed. They added a third parameter which is a boolean.
> 
> ```
> List<SubscriptionInfo> getActiveSubscriptionInfoList(String str, String str2, boolean z) throws RemoteException;
> ```
> 
> The source code of beta versions won't be released until it reaches stable. There is no documentation and I don't know what the boolean means. EDIT: the third parameter is named `isForAllProfiles`, probably user profiles?
> 
> I haven't used hidden apis in Android Studio. So I modified Pixel IMS using APKTool.
> 
> The only file I modified is CarrierModer.smali, starting at line 310.
> 
> ```
> const/4 v1, 0x0
> 
> .line 81
> invoke-interface {v0, p0, p0, v1}, Lcom/android/internal/telephony/ISub;->getActiveSubscriptionInfoList(Ljava/lang/String;Ljava/lang/String;Z)Ljava/util/List;
> ```
> 
> Passing false for the third parameter works (at least for now).
> 
> If you need it you can download it below. GitHub doesn't allow uploading apk files so I put it in a zip archive.
> 
> [pixelims_zipaligned.zip](https://github.com/kyujin-cho/pixel-volte-patch/files/13896231/pixelims_zipaligned.zip)
> 
> or get it on Google Drive: https://drive.google.com/file/d/1zqVShKvd9adwXSFVB1yVh81jjcX4dagB/view?usp=sharing

It works. Thank you

**Comment by ofir29200** [comment]
> @ofir29200 Could you please check if you have the latest version of Shizuku installed (v13.5.3 as of now)? I did just modify the two lines of code I've posted, that shouldn't affect other parts of the app.

i have Shizuku version 13.5.3.r1036.fff3f87

**Comment by iKirby** [comment]
@ofir29200 Please try restarting the app after granting permission. If it's still not working, then I don't know why. I can't reproduce it on both my phones (one with my modified version of Pixel IMS and one with the original version).

**Comment by ofir29200** [comment]
> @ofir29200 Please try restarting the app after granting permission. If it's still not working, then I don't know why. I can't reproduce it on both my phones (one with my modified version of Pixel IMS and one with the original version).

i already tried to restart it. i even tried to clear all of the data of the app, and even reinstall the app, and its still the same....

**Comment by weigesxu** [comment]
Device: pixel 7
System: AP11.231215.007
Shizuku:13.5.3.r1036

Even after using a modified version of the program, it still crashes and cannot enter pixel ims normally.

**Comment by morhimi** [comment]
> Device: pixel 7 System: AP11.231215.007 Shizuku:13.5.3.r1036
> 
> Even after using a modified version of the program, it still crashes and cannot enter pixel ims normally.

are you sure you are using the modified version?
you need to uninstall the original version prior to installing this modified version since they don't have the same signature

**Comment by vard0ger** [comment]
With modified version all work fine.

- Clear all data and remove old version.
- Install mod version.
- Profit.

Px6 14 (AP11.231215.007)

**Comment by morhimi** [comment]
@kyujin-cho do you plan to release a new version?

**Comment by kyujin-cho** [comment]
@morhimi  Eh... I acknowledged what's the problem. But for now, as @iKirby mentioned - on current way of implementation, until we see the official release of android software (along with published source code) we have not much things to do. Let me see if there's a workaround.

**Comment by kyujin-cho** [comment]
Should be resolved by #231.
